src/test/java/com/example/postsapi/mq/SenderTest.java

```diff
@@ -0,0 +1,56 @@
+package com.example.postsapi.mq;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.boot.test.mock.mockito.MockBean;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doNothing;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SenderTest {
+
+    @Mock
+    private Sender sender;
+
+    @Mock
+    private RabbitTemplate rabbitTemplate;
+
+    private final ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+    private final ByteArrayOutputStream errContent = new ByteArrayOutputStream();
+    private final PrintStream originalOut = System.out;
+    private final PrintStream originalErr = System.err;
+
+    @Before
+    public void setUpStreams() {
+        System.setOut(new PrintStream(outContent));
+        System.setErr(new PrintStream(errContent));
+    }
+
+    @After
+    public void restoreStreams() {
+        System.setOut(originalOut);
+        System.setErr(originalErr);
+    }
+
+    @Test
+    public void sendPostId_Success() {
+        sender.sendPostId("1");
+
+        System.out.println("PostId sent: 1 on q: queue1");
+
+        assertEquals("PostId sent: 1 on q: queue1\n", outContent.toString());
```

Comment:
wait what?  is outContent supposed to be a stream from the sender.sendPostId or is it grabbing from the sout above it?

Reply:
Should be the sender.sendPostId...might not..